src/nrnpython/nrnpy_hoc.cpp

```diff
@@ -1701,7 +1701,6 @@ static PyObject* iternext_sl(PyHocObject* po, hoc_Item* ql) {
       return NULL;
     }
   } else if (po->u.its_ == PyHoc::NextNotLast) {
-    nrn_popsec(); // pop the previous iteration.
```

Comment:
I assume this is intentional? Was not obvious from the title of the PR?

Reply:
Yes, intentional. That whole function no longer does a popsec. Surprised when I saw it but it was an edge case.